app/src/org/commcare/views/EntityDetailView.java

```diff
@@ -320,6 +330,21 @@ public boolean onTouch(View view, MotionEvent event) {
 
             graphLayout.removeAllViews();
             graphLayout.addView(graphView, GraphView.getLayoutParams());
+            if (showSpinner) {
+                final ProgressBar spinner = new ProgressBar(this.getContext(), null, android.R.attr.progressBarStyleLarge);
+                spinner.setLayoutParams(new FrameLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT, Gravity.CENTER));
+                GraphLoader graphLoader = new GraphLoader((Activity) this.getContext(), spinner);

```

Comment:
It wouldn't be a huge deal to only run this on 18+, but yeah, I think it's not an issue because it's all local js we have control over.